packages/jest-cli/src/runJest.js

```diff
@@ -79,54 +90,69 @@ const runJest = (
               );
             }
           }
-
+        }
+        if (!data.benchPaths || !data.benchPaths.length) {
+          noBenchesFound = true;
+        }
+        if (noTestsFound && noBenchesFound) {
           localConsole.log(
-            source.getNoTestsFoundMessage(patternInfo, config, data),
+            source.getNoTestsOrBenchmarksFoundMessage(
+              patternInfo,
+              config,
+              data,
+              noTestsFound,
+              noBenchesFound,
+            ),
           );
         }
         return data;
       }).then(data => {
-        if (data.paths.length === 1) {
+        if (data.testPaths.length === 1) {
           if (config.silent !== true && config.verbose !== false) {
             // $FlowFixMe
             config = Object.assign({}, config, {verbose: true});
           }
         }
-
-        return new TestRunner(
-          hasteContext,
-          config,
-          {
-            getTestSummary: () => getTestSummary(argv, patternInfo),
-            maxWorkers,
-          },
-          startRun,
-        ).runTests(data.paths, testWatcher);
+        return Promise.all([
+          new TestRunner(
+            hasteContext,
+            config,
+            {
+              getTestSummary: () => getTestSummary(argv, patternInfo),
+              maxWorkers,
+            },
+            startRun,
+          ).runTests(data.testPaths, testWatcher),
+          new BenchmarkRunner(
+            hasteContext,
+            config,
+          ).runBenches(data.benchPaths),
+        ]);
```

Comment:
I added this to the issues/points section above. I wasn't sure and wanted some feedback before implementing that feature :)